### DISCUSSION TITLE
useHydrate with Provider and Store don't reset on unmount

### DISCUSSION THREAD

**Discussion by jarandmi** [pmndrs/jotai#2220] [Bug report]
I expect the store to be reset when renderKey is changed. This only works when i don't have a store connected to the Provider. Is this intended? If yes, how can i hydrate/reset all atoms the same way that I can without the store prop

Example:
```javascript
const store = createStore()

const Root =({ renderKey })=>{
  <Provider store={store} key={renderKey}>
    <Hydrate />
  </Provider>
}
```
```javascript
const Hydrate = ()=> {
  useHydrateAtoms([[dataAtom, crypto.randomUUID()]])
  return null
}
```

This will not reset/hydrate when the `renderKey`prop changes

**Comment by dai-shi** [comment]
Yes, you need to create a new store for different `renderKey`. It's intended.

**Reply by BigDog1400** [reply]
Do you have an example an example of how to do it?

**Reply by himself65** [reply]
create a function similar to atomFamily, Let's call it `storeFamily`

```ts
const storeFamily = () => {
  const map = {}
  return (key) => {
    if (!map[key]) {
      map[key] = createStore()
    }
    return map[key]
  }
}

const family = storeFamily()

const Root =({ renderKey })=>{
  <Provider store={family(renderKey)} >
    <Hydrate />
  </Provider>
}
```
